template/src/ontology/run.sh.jinja2

```diff
@@ -90,6 +90,22 @@ rm -f tmp/debug.log
 VOLUME_BIND=$PWD/../../:/work$ODK_SSH_BIND
 WORK_DIR=/work/src/ontology
 
+# Support for OAK cache sharing
+if [ -n "$ODK_SHARE_OAK_CACHE" ]; then
+    case "$ODK_SHARE_OAK_CACHE" in
+    user)
+        # We assume the cache is in its default location; if it is not,
+        # ODK_SHARE_OAK_CACHE must point to the actual location.
+        ODK_SHARE_OAK_CACHE="$HOME/.data/oaklib"
```

Comment:
We _could_ in fact not assume anything, and automatically find the user’s cache no matter where it is located (that’s what the ODKRunner does).

But it would be quite longer (not really complex, just longer), because there are many ways to change the default location of the cache and we’d need to check them all.

This would look like this:

```sh
if [ -n "$ODK_SHARE_OAK_CACHE" ]; then
    case "$ODK_SHARE_OAK_CACHE" in
    user)
        if [ -n "$OAKLIB_HOME" ]; then
            ODK_SHARE_OAK_CACHE="$OAKLIB_HOME"
        elif [ -n "$PYSTOW_HOME" ]; then
            ODK_SHARE_OAK_CACHE="$PYSTOW_HOME/oaklib"
        elif [ -n "$PYSTOW_USE_APPDIRS" ]; then
            if [ -n "$XDG_DATA_HOME" ]; then
                ODK_SHARE_OAK_CACHE="$XDG_DATA_HOME/oaklib"
            elif [ -d "$HOME/Library/Application Support" ]; then
                ODK_SHARE_OAK_CACHE="$HOME/Library/Application Support/oaklib";
            else
                ODK_SHARE_OAK_CACHE="$HOME/.local/share/oaklib"
            fi
        else
            ODK_SHARE_OAK_CACHE="$HOME/${PYSTOW_NAME:-.data}/oaklib"
        fi
        ;;
```

I am not sure whether it brings enough benefits to justify making the `run.sh` script so long.

Reply:
What is your concern with the length of the script: Unless the code introduces syntax that might cause some kinds of shells to choke (i.e. non POSIX) I don't see run.sh length as much of an issue anymore.

Reply:
I am simply concerned that this script is turning into a huge piece of bloat, that is becoming harder and harder to maintain (for us) and to understand (for everybody else).